pkg/cmd/logs.go

```diff
@@ -197,11 +201,11 @@ func logPod(toMatch string, toTarget string, container string) {
 	namespace := getSeedNamespaceNameForShoot(target.Target[2].Name)
 	var err error
 
-	project, err := getProjectForShoot()
+	project, err := getProjectObject()
 	checkError(err)
-	shootName := target.Target[2].Name
-
-	shoot, err := Client.GardenerV1beta1().Shoots(*project.Spec.Namespace).Get(shootName, metav1.GetOptions{})
+	shootName, err := getTargetName("shoot")
+	checkError(err)
+	shoot, err := getShootObject(*project.Spec.Namespace, shootName)
```

Comment:
> yes Planning use that getShootObject() call from any other place for the different namespace and/or shootname and extend metav1.GetOptions if need in the future. Shall I leave in miscellaneous.go or become one of TargetInterface in types.go ?

not sure what you mean exactly, maybe you can give an example.
However I would prefer to have a simple `getShootObject()` and If you still want to have a `getShootObject(namespace, name)` function you can introduce it later and call it from the getShootObject() function. I make an example so that we do not have a misunderstanding :)

Discalimer: code not tested below, but you get the idea
```go
func getShootObjectFromTarget() (*v1beta1.Shoot, error) {
	project, err := getProjectObject()
	if err != nil {
		return nil, err
	}
	shootName, err := getTargetName("shoot")
	if err != nil {
		return nil, err
	}

	return getShootObject(*project.Spec.Namespace, shootName)
}

func getShootObject(namespace string, name string) (*v1beta1.Shoot, error) {
	var target Target
	ReadTarget(pathTarget, &target)
	gardenClientset, err := target.GardenerClient()
	checkError(err)
	shoot, err := gardenClientset.CoreV1beta1().Shoots(namespace).Get(name, metav1.GetOptions{})
	checkError(err)
	return shoot, nil
}
```

But I guess for now you can have it in one function. Currently I do not see a need for an own `getShootObject(namespace string, name string)`

Reply:
Thanks, I got it. I was thinking getShootObject more independence and decoupling without call `getProjectObject()`  when I design.  also the case about `garden` namespace for seed without query. that why I use `namespace string, name string` input.  but I think that will take care by getTargetName("project") to return `garden`  as well 

 I will change it.

Reply:
fixed